src/Workspaces/Core/Portable/PatternMatching/PatternMatcher.cs

```diff
@@ -166,112 +147,101 @@ private static bool ContainsUpperCaseLetter(string pattern)
             TextChunk patternChunk,
             bool punctuationStripped)
         {
-            int caseInsensitiveIndex = _compareInfo.IndexOf(candidate, patternChunk.Text, CompareOptions.IgnoreCase);
+            var candidateLength = candidate.Length;
+
+            var caseInsensitiveIndex = _compareInfo.IndexOf(candidate, patternChunk.Text, CompareOptions.IgnoreCase);
             if (caseInsensitiveIndex == 0)
             {
-                if (patternChunk.Text.Length == candidate.Length)
+                // We found the pattern at the start of the candidate.  This is either an exact or
+                // prefix match. 
+
+                if (patternChunk.Text.Length == candidateLength)
                 {
-                    // a) Check if the part matches the candidate entirely, in an case insensitive or
-                    //    sensitive manner.  If it does, return that there was an exact match.
+                    // Lengths were the same, this is either a case insensitive or sensitive exact match.
                     return new PatternMatch(
                         PatternMatchKind.Exact, punctuationStripped, isCaseSensitive: candidate == patternChunk.Text,
-                        matchedSpan: GetMatchedSpan(0, candidate.Length));
+                       matchedSpan: GetMatchedSpan(0, candidateLength));
                 }
                 else
                 {
-                    // b) Check if the part is a prefix of the candidate, in a case insensitive or sensitive
-                    //    manner.  If it does, return that there was a prefix match.
+                    // Lengths were the same, this is either a case insensitive or sensitive prefix match.
                     return new PatternMatch(
                         PatternMatchKind.Prefix, punctuationStripped, isCaseSensitive: _compareInfo.IsPrefix(candidate, patternChunk.Text),
                         matchedSpan: GetMatchedSpan(0, patternChunk.Text.Length));
                 }
             }
 
-            var isLowercase = patternChunk.IsLowercase;
-            if (isLowercase)
+            ArrayBuilder<TextSpan> candidateHumpsOpt = null;
```

Comment:
If we end up creating this in one path, we store it so we can use it if needed later.  this avoids creating the data multiple times.